internal/connector/internal/services/connector_namespaces.go

```diff
@@ -312,7 +313,7 @@ func (k *connectorNamespaceService) GetExpiredNamespaceIds() ([]string, *errors.
 	dbConn := k.connectionFactory.New()
 	now := time.Now()
 	if err := dbConn.Model(&dbapi.ConnectorNamespace{}).Select("id").
-		Where("expiration < ?", now).Find(&result).Error; err != nil {
+		Where("expiration < ? AND status_phase NOT IN ?", now, []string{string(dbapi.ConnectorNamespacePhaseDeleting), string(dbapi.ConnectorNamespacePhaseDeleted)}).Find(&result).Error; err != nil {
```

Comment:
We also need to exclude `deleting` and `deleted` namespaces in another update https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/4a31a0543d271107e97fd74482bd870afefc2dac/internal/connector/internal/services/connector_namespaces.go#L389

Reply:
This should fix the not yet seen bug that the test approach you proposed should expose. Correct?

Reply:
yes